frontend/src/pages/course/components/LeaveCourse/LeaveCourseButton.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC, useState } from "react";
+import { Button, message, Modal, Typography } from "antd";
+import { leaveCourse, LeaveStatus } from "./CourseMembershipService";
+import { useTranslation } from "react-i18next";
+
+interface LeaveCourseButtonProps {
+    courseId: string;
+}
+
+const LeaveCourseButton: FC<LeaveCourseButtonProps> = ({ courseId }) => {
+    const { Text } = Typography;
+    const { t } = useTranslation();
+    const [confirmLeaveVisible, setConfirmLeaveVisible] = useState<boolean>(false);
+
+    const handleLeaveConfirm = async () => {
+        const result = await leaveCourse(courseId, t); // Pass the translation function as a parameter
+        if (result.success) {
+            message.success(result.message);
+            setTimeout(() => {
+                window.location.href = '/';
```

Comment:
Gebruik de `useNavigate` hook om naar een andere pagina te navigeren. Als je via `window.location.href` werkt, dan moet de browser de hele website terug opnieuw downloaded (wat vrij traag is) wat niet de bedoeling is voor een single page application.

Reply:
Ok, ik heb dit aangepast